### ISSUE TITLE
Test test_nanmedian_all and test_nanmedian_all_axis sometimes fails (side effects?)

### ISSUE THREAD

**Issue by joernhees** [scipy/scipy#5656]
I recently upgraded python to 2.7.11 with brewed python on Mac OS X 10.10.5, NumPy to 1.10.2 and SciPy to 0.16.1 ... running the tests from ipython with `numpy.test()` and `scipy.test()` i got a failure in `test_nanmedian_all`.

Re-running just the scipy tests in a plain python shell they first passed... re-running them failed again. I can re-produce this with the following:

```python
$ python
Python 2.7.11 (default, Dec 26 2015, 17:47:15)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import scipy
>>> scipy.test('test_nanmedian_all')
Running unit tests for scipy
NumPy version 1.10.2
NumPy relaxed strides checking option: False
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
SciPy version 0.16.1
SciPy is installed in /usr/local/lib/python2.7/site-packages/scipy
Python version 2.7.11 (default, Dec 26 2015, 17:47:15) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
nose version 1.3.7
/usr/local/lib/python2.7/site-packages/numpy/lib/utils.py:99: DeprecationWarning: `scipy.weave` is deprecated, use `weave` instead!
  warnings.warn(depdoc, DeprecationWarning)
...............
----------------------------------------------------------------------
Ran 15 tests in 2.032s

OK
<nose.result.TextTestResult run=15 errors=0 failures=0>
>>> scipy.test('test_nanmedian_all')
Running unit tests for scipy
NumPy version 1.10.2
NumPy relaxed strides checking option: False
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
SciPy version 0.16.1
SciPy is installed in /usr/local/lib/python2.7/site-packages/scipy
Python version 2.7.11 (default, Dec 26 2015, 17:47:15) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
nose version 1.3.7
....FF.........
======================================================================
FAIL: test_nanmedian_all (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/scipy/stats/tests/test_stats.py", line 217, in test_nanmedian_all
    assert_equal(len(w), 2)  # Deprecation & RuntimeWarning
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 355, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 0
 DESIRED: 2

======================================================================
FAIL: test_nanmedian_all_axis (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/scipy/stats/tests/test_stats.py", line 226, in test_nanmedian_all_axis
    assert_equal(len(w), 4)
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 355, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 3
 DESIRED: 4

----------------------------------------------------------------------
Ran 15 tests in 1.141s

FAILED (failures=2)
<nose.result.TextTestResult run=15 errors=0 failures=2>
...
```

subsequent invocations of the test seem to fail, running them in a new python shell, the first invocation will pass, the following ones will fail.

I also tried first running the numpy tests and then just the failing scipy tests as shown below:

```python
$ python
Python 2.7.11 (default, Dec 26 2015, 17:47:15)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.test()
Running unit tests for numpy
NumPy version 1.10.2
NumPy relaxed strides checking option: False
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
Python version 2.7.11 (default, Dec 26 2015, 17:47:15) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
nose version 1.3.7
................................................................................................................................................................................................................................................................................................................................../usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py:473: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  callableObj(*args, **kwargs)
......../usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py:473: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  callableObj(*args, **kwargs)
/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py:473: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  callableObj(*args, **kwargs)
...............................................................................................................................................................................................................................................................................................................................S..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K........................................................./usr/local/lib/python2.7/site-packages/numpy/ma/core.py:989: RuntimeWarning: invalid value encountered in multiply
  masked_da = umath.multiply(m, da)
/usr/local/lib/python2.7/site-packages/numpy/ma/core.py:989: RuntimeWarning: invalid value encountered in multiply
  masked_da = umath.multiply(m, da)
........................................................................................................................................../usr/local/lib/python2.7/site-packages/numpy/core/tests/test_numerictypes.py:372: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  return self.ary['f0', 'f1']
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K...SK.S.......S...................................................................................clang: warning: -Wl,-rpath=/var/folders/95/hl31m68d6ks1d8txd27wxq000000gn/T/tmpeWnQYH/libbar.so: 'linker' input unused
..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S......................................................................................................................S............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 5931 tests in 17.918s

OK (KNOWNFAIL=3, SKIP=7)
<nose.result.TextTestResult run=5931 errors=0 failures=0>
>>> import scipy
>>> scipy.test('test_nanmedian_all')
Running unit tests for scipy
NumPy version 1.10.2
NumPy relaxed strides checking option: False
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
SciPy version 0.16.1
SciPy is installed in /usr/local/lib/python2.7/site-packages/scipy
Python version 2.7.11 (default, Dec 26 2015, 17:47:15) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
nose version 1.3.7
/usr/local/lib/python2.7/site-packages/numpy/lib/utils.py:99: DeprecationWarning: `scipy.weave` is deprecated, use `weave` instead!
  warnings.warn(depdoc, DeprecationWarning)
.....F.........
======================================================================
FAIL: test_nanmedian_all_axis (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/scipy/stats/tests/test_stats.py", line 226, in test_nanmedian_all_axis
    assert_equal(len(w), 4)
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 355, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 1
 DESIRED: 4

----------------------------------------------------------------------
Ran 15 tests in 1.981s

FAILED (failures=1)
<nose.result.TextTestResult run=15 errors=0 failures=1>
>>> scipy.test('test_nanmedian_all')
Running unit tests for scipy
NumPy version 1.10.2
NumPy relaxed strides checking option: False
NumPy is installed in /usr/local/lib/python2.7/site-packages/numpy
SciPy version 0.16.1
SciPy is installed in /usr/local/lib/python2.7/site-packages/scipy
Python version 2.7.11 (default, Dec 26 2015, 17:47:15) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
nose version 1.3.7
....FF.........
======================================================================
FAIL: test_nanmedian_all (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/scipy/stats/tests/test_stats.py", line 217, in test_nanmedian_all
    assert_equal(len(w), 2)  # Deprecation & RuntimeWarning
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 355, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 0
 DESIRED: 2

======================================================================
FAIL: test_nanmedian_all_axis (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/scipy/stats/tests/test_stats.py", line 226, in test_nanmedian_all_axis
    assert_equal(len(w), 4)
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 355, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 0
 DESIRED: 4

----------------------------------------------------------------------
Ran 15 tests in 1.107s

FAILED (failures=2)
<nose.result.TextTestResult run=15 errors=0 failures=2>
>>>
```

as you can see running `numpy.test()` seems to only cause one of the tests to fail, while re-running causes both to fail.

**Comment by rgommers** [comment]
This is due to how Python warnings behave - they get shown only once by default. So inside a shell the second time no warning is shown. There are more numpy and scipy tests that fail this way - is a wontfix I think. Either way, ``nanmedian`` is deprecated, so will send a PR to remove it right now. Then the problem is definitely gone.

**Comment by joernhees** [comment]
oh, ok. thanks for the explanation though ;)

**Comment by agrapentin** [comment]
This test is running and failing for me in scipy 0.17.0 with numpy 1.11.0b3 and python 3.5.1 compiled with gcc 4.8.4 and the latest mkl. I note above it is depreciated, however just stating the test is still there and failing.

```console
$ pip show scipy | grep ^Version:
Version: 0.17.0
$ pip show numpy | grep ^Version:
Version: 1.11.0b3
$ python -c "import scipy; scipy.test(label='test_nanmedian_all');"
Running unit tests for scipy
NumPy version 1.11.0b3
NumPy relaxed strides checking option: False
NumPy is installed in /home/andrew/local/python35/lib/python3.5/site-packages/numpy
SciPy version 0.17.0
SciPy is installed in /home/andrew/local/python35/lib/python3.5/site-packages/scipy
Python version 3.5.1 (default, Feb 13 2016, 22:48:20) [GCC 4.8.4]
nose version 1.3.7
.....F.........
======================================================================
FAIL: test_nanmedian_all_axis (test_stats.TestNanFunc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andrew/local/python35/lib/python3.5/site-packages/scipy/stats/tests/test_stats.py", line 242, in test_nanmedian_all_axis
    assert_equal(len(w), 4)
  File "/home/andrew/local/python35/lib/python3.5/site-packages/numpy/testing/utils.py", line 375, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
 ACTUAL: 5
 DESIRED: 4

----------------------------------------------------------------------
Ran 15 tests in 2.920s

FAILED (failures=1)
```

**Comment by rgommers** [comment]
@agrapentin thanks for the report. The function was removed from master a month ago, it'll not be present in 0.18.0 anymore.
